### PR TITLE
✨(agent): Implement QA agent schema issue auto-repair mechanism

### PR DESCRIPTION
**Pull request by sasamuku**
## Issue

- resolve: https://github.com/route06/liam-internal/issues/5965

## Why is this change needed?

When test cases fail in the QA agent, we needed a mechanism to automatically distinguish whether the failure is due to SQL issues or schema issues, and handle them appropriately.

Previously, all failures were treated as SQL issues, but in reality, some failures require schema-side fixes (such as missing tables or columns). Retrying on the SQL side without distinguishing these cases cannot resolve the problems and leads to a degraded user experience.

### Key Features

1. **LLM-based Failure Classification**
   - Uses OpenAI o4-mini to automatically classify test failures
   - `SQL_ISSUE`: Issues with the SQL query itself (syntax errors, type mismatches, etc.)
   - `SCHEMA_ISSUE`: Issues with the schema (missing tables/columns, missing constraints, etc.)

2. **Priority-based Routing**
   - SQL issues: Prioritized for retry (up to `QA_AGENT_MAX_ATTEMPTS` times) as they can be fixed quickly
   - Schema issues: **Always routed to DB Agent** (no retry limit)
   - Schema problems require structural modifications, so they are delegated to the specialized DB Agent

3. **Cross-thread Persistence**
   - Integrates with DB Agent through `schemaIssues` annotation
   - Converts schema problems to `SchemaIssue` format for handoff

### Implementation Details

**New Files:**
- `analyzeTestFailuresNode.ts`: LLM-based failure classification logic
- `convertToSchemaIssuesNode.ts`: Converts schema problems to SchemaIssue format
- `routeAfterAnalyzeFailures.ts`: Routes based on failure type and retry count

**Key Changes:**
- `createQaAgentGraph.ts`: Integrated analysis and routing nodes after executeTestcases
- `constants.ts`: Added `QA_AGENT_MAX_ATTEMPTS` constant (CI: 1 attempt, Local: 3 attempts)

**Tests:**
- Unit tests verifying behavior of each node
- Integration tests using real OpenAI API (excluded from CI via `*.integration.test.ts` pattern)

### Graph Flow

```
executeTestcases
    ↓
analyzeTestFailures (LLM classification)
    ↓
routeAfterAnalyzeFailures
    ├→ resetFailedSqlTests (SQL issues with retry available)
    ├→ convertToSchemaIssues (Schema issues detected)
    └→ END (No failures or SQL retry limit reached)
```

### Important Design Decisions

- **Schema issues are exempt from retry limits**: Schema problems cannot be resolved by SQL retries, so they are always sent to DB Agent
- **Environment-dependent configuration**: CI environment has no retries (1 attempt only) for fast execution, development environment allows 2 retries (3 attempts total)
- **Simple implementation**: Avoids complex branching, implements retry logic with direct comparison against `QA_AGENT_MAX_ATTEMPTS`

## Test Coverage

- **5 integration tests** for failure classification (with real OpenAI API, excluded from CI)
- **4 unit tests** for schema issue conversion
- **6 unit tests** for routing logic
- **1 unit test** for graph integration
- **Total: 16 tests** covering all scenarios and edge cases

All tests pass with proper type safety.

## Release Notes

* **New Features**
  * Enhanced test failure analysis with intelligent classification into SQL and schema issue categories for improved troubleshooting
  * Added schema issue handling workflow to the QA agent pipeline

* **Tests**
  * Added comprehensive test coverage for test failure analysis and schema issue conversion workflows

* **Chores**
  * Updated internal QA agent retry configuration and routing logic

### CODE REVIEW

File: frontend/internal-packages/agent/src/qa-agent/nodes/analyzeTestFailuresNode.ts [hunk 1]
```diff
@@ -13,33 +60,84 @@ const extractFailedTests = (
...
-      // TODO: Implement schema vs SQL issue classification
-      failureAnalysis: {
-        failedSqlTestIds: failedTestIds,
-        failedSchemaTestIds: [],
-      },
-    }),
+    failureAnalysis: {
+      failedSqlTestIds: sqlIssueIds,
+      failedSchemaTestIds: schemaIssueIds,
+    },
```
[the remark this answers is not in the document]
Comment (sasamuku) [reply frontend/internal-packages/agent/src/qa-agent/nodes/analyzeTestFailuresNode.ts]: @claude 
fix this
